apply-major-version-tags

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env ruby
```

Comment:
I was expecting this commit to also delete some code from the mirror script, but maybe that comes later in the PR?

Reply:
Yes, the intention of this PR is just to get the script onto GitHub so we can let it run in dry-run mode a few times, leaving the mirror script as-is. Then once we're happy this works as expected, we can remove the tagging code from the mirror script and switch this script out of dry-run mode.